packages/smooth_app/lib/query/product_query.dart

```diff
@@ -53,29 +53,64 @@ abstract class ProductQuery {
   }
 
   /// Returns the global country for API queries.
-  static OpenFoodFactsCountry? getCountry() => _country;
+  static OpenFoodFactsCountry getCountry() => _country;
+
+  /// Sets the global country for API queries: implicit choice at init time.
+  static Future<void> initCountry(
+    final UserPreferences userPreferences,
+  ) async {
+    const OpenFoodFactsCountry defaultCountry =
+        OpenFoodFactsCountry.SWITZERLAND; // why not?
```

Comment:
I think this one will be the subject to discussions

Reply:
We need a country.
The normal behavior is to use the device's country.
If everything falls apart, only then will we use the default country. And the user sees in the onboarding that it can be changed to what is more relevant.

I have no problem changing the default country (I would even say "fallback country"). Just pick a "more relevant" country.

Reply:
Yes, that was exactly, the idea behind my comment.


What's the best value to use @teolemon and @stephanegigandet?

Reply:
In which case could it be null ? In that case, we should have a special screen appearing, asking to fix the issue ?

Reply:
It would be null only in once case: when the app is started the very first time, we need to pre-select the country of the user. For that we use some platform data (`PlatformDispatcher.instance.locale.countryCode`). If this platform data is not available or is a code that does not match "our" countries, the country will be null.
Only in that case will we use the default country (so far, Switzerland).
And in the onboarding, the user will see the pre-selected country and will have the obvious possibility to change it.

Reply:
I'm proposing France, since we have contributors monitoring the influx of new products that could be a consequence of that